src/sgidClient.ts

```diff
@@ -0,0 +1,233 @@
+      this.privateKey = convertPkcs1ToPkcs8(privateKey)
+    } else {
+      this.privateKey = privateKey
+    }
+
+    /**
+     * To rectify instances where an extra '\' escape character is added to the newline
+     */
+    this.privateKey = this.privateKey.replace(/\\n/gm, '\n')
+  }
+
+  /**
+   * Generates authorization url to redirect end-user to sgID login page.
+   * @param state A string which will be passed back to your application once the end-user
+   * logs in. You should use this to prevent cross-site request forgery attacks (see
+   * https://www.rfc-editor.org/rfc/rfc6749#section-10.12). You can also use this to
+   * track per-request state.
+   * @param scope Array or space-separated scopes. 'openid' must be provided as a scope.
+   * Defaults to 'myinfo.nric_number openid'.
+   * @param nonce Unique nonce for this request. If this param is undefined, a nonce is generated
+   * and returned. To prevent this behaviour, specify null for this param.
+   * @param redirectUri The redirect URI used in the authorization request. If this param is provided,
+   * it will be used over the redirect URI provided in the SgidClient constructor. The redirect URI
+   * must be provided in at least the SgidClient constructor or this function.
+   * @param codeChallenge The code challenge from the code verifier used for PKCE enhancement
+   */
+  authorizationUrl({
+    state,
+    scope = DEFAULT_SCOPE,
+    nonce = generators.nonce(),
+    redirectUri = this.getFirstRedirectUri(),
+    codeChallenge,
+  }: AuthorizationUrlParams): AuthorizationUrlReturn {
+    const url = this.sgID.authorizationUrl({
+      scope: typeof scope === 'string' ? scope : scope.join(' '),
+      nonce: nonce ?? undefined,
+      state,
+      redirect_uri: redirectUri,
+      code_challenge: codeChallenge,
+      code_challenge_method: DEFAULT_SGID_CODE_CHALLENGE_METHOD,
+    })
+
+    const result: { url: string; nonce?: string } = { url }
+    if (nonce) {
+      result.nonce = nonce
+    }
+    return result
+  }
+
+  private getFirstRedirectUri(): string {
+    if (
+      !this.sgID.metadata.redirect_uris ||
+      this.sgID.metadata.redirect_uris.length === 0
+    ) {
+      throw new Error(Errors.MISSING_REDIRECT_URI_ERROR)
+    }
+    return this.sgID.metadata.redirect_uris[0]
+  }
+
+  /**
+   * Exchanges authorization code for access token.
+   * @param code The authorization code received from the authorization server
+   * @param nonce Nonce passed to authorizationUrl for this request. Specify null
+   * if no nonce was passed to authorizationUrl.
+   * @param redirectUri The redirect URI used in the authorization request. Defaults to the one
+   * passed to the SgidClient constructor.
+   * @param codeVerifier The code verifier that was used to generate the code challenge that was passed in `authorizationUrl`
+   * @returns The sub (subject identifier claim) of the user and access token. The subject
```

Comment:
returns the access token too, no?

Reply:
yup it returns the access token too - the full comment is `The sub (subject identifier claim) of the user and access token. The subject identifier claim is the end-user's unique ID.`

do you think it's sufficient or can we make it clearer?

Reply:
oops lol clearly I misread, my bad. current version makes sense!